bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/models/AbstractScalarResponse.java

```diff
@@ -0,0 +1,160 @@
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    public <T> T as(final Class<T> clazz) throws IOException {
+        Objects.requireNonNull(clazz, "clazz cannot be null");
+
+        // First see if there is a constructor that takes a ScalarWebResult (us)
+        // If so - call it with us
+        // Otherwise try to use GSON to construct the class and set the fields
+        try {
+            final Constructor<T> constr = clazz.getConstructor(this.getClass());
+            return constr.newInstance(this);
+        } catch (final NoSuchMethodException e) {
+            final JsonArray localResults = getPayload();
+            if (localResults == null || isBlank(localResults)) {
+                throw new IllegalArgumentException(
+                        "Cannot convert ScalarWebResult for " + clazz + " with results: " + localResults);
+            } else if (localResults.size() == 1) {
+                JsonElement elm = localResults.get(0);
+                if (elm.isJsonArray()) {
+                    final JsonArray arry = elm.getAsJsonArray();
+                    if (arry.size() == 1) {
+                        elm = arry.get(0);
+                    } else {
+                        elm = arry;
+                    }
+                }
+                final Gson gson = GsonUtilities.getApiGson();
+
+                if (elm.isJsonObject()) {
+                    final JsonObject jobj = elm.getAsJsonObject();
+                    return gson.fromJson(jobj, clazz);
+                } else {
+                    return gson.fromJson(elm, clazz);
+                }
+            }
+            throw new IllegalArgumentException(
+                    "Cannot convert ScalarWebResult to " + clazz + " with results: " + localResults, e);
+
+        } catch (InstantiationException | IllegalAccessException | IllegalArgumentException
+                | InvocationTargetException e) {
+            throw new IllegalArgumentException(
+                "Cannot convert ScalarWebResult to " + clazz + " for reason: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Converts this generic response into an array of the specified type
+     *
+     * @param <T> the generic type that will be returned
+     * @param clazz the class to cast to
+     * @return a non-null, possibly empty list of objects converted to the class
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    public <T> List<T> asArray(final Class<T> clazz) throws IOException {
+        Objects.requireNonNull(clazz, "clazz cannot be null");
+
+        final JsonArray localResults = getPayload();
+        if (localResults == null) { // empty array is okay here - just not null
+            throw new IllegalArgumentException(
+                    "Cannot convert ScalarWebResult for " + clazz + " with results: " + localResults);
+        }
+        
+        final Gson gson = new Gson();
```

Comment:
Can a shared instance of Gson be used instead?

Reply:
@Hilbrand already mentioned this one and done